ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/load/table/LoadPartitions.java

```diff
@@ -304,33 +327,33 @@ private TaskTracker forExistingTable(AddPartitionDesc lastPartitionReplicated) t
     while (partitionIterator.hasNext() && tracker.canAddMoreTasks()) {
       AddPartitionDesc addPartitionDesc = partitionIterator.next();
       Map<String, String> partSpec = addPartitionDesc.getPartition(0).getPartSpec();
-      Partition ptn = context.hiveDb.getPartition(table, partSpec, false);
-      if (ptn == null) {
-        if (!replicationSpec.isMetadataOnly()) {
-          addPartition(partitionIterator.hasNext(), addPartitionDesc);
-        }
-      } else {
-        // If replicating, then the partition already existing means we need to replace, maybe, if
-        // the destination ptn's repl.last.id is older than the replacement's.
-        if (replicationSpec.allowReplacementInto(ptn.getParameters())) {
-          if (replicationSpec.isMetadataOnly()) {
-            tracker.addTask(alterSinglePartition(addPartitionDesc, replicationSpec, ptn));
-            if (!tracker.canAddMoreTasks()) {
-              tracker.setReplicationState(
-                  new ReplicationState(
-                      new PartitionState(table.getTableName(), addPartitionDesc)
-                  )
-              );
-            }
-          } else {
-            addPartition(partitionIterator.hasNext(), addPartitionDesc);
-          }
-        } else {
-          // ignore this ptn, do nothing, not an error.
-        }
+      Task<?> ptnRootTask = null;
+      ReplLoadOpType loadPtnType = getLoadPartitionType(partSpec);
+      switch (loadPtnType) {
+        case LOAD_NEW:
+          break;
+        case LOAD_REPLACE:
+          ptnRootTask = dropPartitionTask(table, partSpec);
+          break;
+        case LOAD_SKIP:
+          continue;
+        default:
+          break;
       }
+      addPartition(partitionIterator.hasNext(), addPartitionDesc, ptnRootTask);
     }
     return tracker;
   }
+
+  private ReplLoadOpType getLoadPartitionType(Map<String, String> partSpec) throws InvalidOperationException, HiveException {
+    Partition ptn = context.hiveDb.getPartition(table, partSpec, false);
+    if (ptn == null) {
+      return ReplLoadOpType.LOAD_NEW;
+    }
+    if (ReplUtils.replCkptStatus(tableContext.dbNameToLoadIn, ptn.getParameters(), context.dumpDirectory)) {
```

Comment:
can this be renamed to isReplCkptPropertySet

Reply:
CkptStatus means if checkpointed or not. So, I think, it gives the same meaning. If you insist and if making another patch, then will rename it.